kion/internal/kionclient/helper.go

```diff
@@ -310,40 +268,64 @@ func OptionalInt(d *schema.ResourceData, fieldname string) *int {
 	if !ok {
 		return nil
 	}
+	return &ret
+}
+
+// OptionalString retrieves a string value from schema.ResourceData by its field name and
+// returns a pointer to that value. If the field is not set or is not a string, it returns nil.
+func OptionalString(d *schema.ResourceData, fieldname string) *string {
+	v, ok := d.GetOkExists(fieldname)
+	if !ok {
+		return nil
+	}
+
+	ret, ok := v.(string)
+	if !ok {
+		return nil
+	}
```

Comment:
Could you use OptionalValue[string] below instead of having this function?

Reply:
Done.